src/AdoNetCore.AseClient/Internal/SchemaTableBuilder.cs

```diff
@@ -163,28 +168,45 @@ private void TryLoadKeyInfo(DataTable table, string baseTableNameValue, string b
 
                 try
                 {
+                    var columnMetadata = new Dictionary<string, ColumnMetadata>();
                     using (var keyInfoDataReader = command.ExecuteReader())
                     {
                         while (keyInfoDataReader.Read())
                         {
-                            var keyColumnName = keyInfoDataReader["COLUMN_NAME"].ToString();
-                            var keySchemaName = keyInfoDataReader["TABLE_SCHEMA"].ToString();
-                            var keyCatalogName = keyInfoDataReader["TABLE_CATALOG"].ToString();
+                            var key = keyInfoDataReader["COLUMN_NAME"].ToString();
+
+                            // Check for duplicate name and if we found any then give it a miss
+                            if (columnMetadata.ContainsKey(key))
```

Comment:
Is there any spec or standard behaviour in the reference driver that indicates we can always take the first record for each column_name? If so we should comment it here.